### PR TITLE
[Exec](expr) Opt the compound pred performace

### PR DESCRIPTION
**Pull request by HappenLee**
### What problem does this PR solve?

before：
```
 mysqlslap -hd3 -uroot -P9130  --create-schema=test_db2 -c 10 -i 500 -q "SELECT count(k) FROM sbtest1_dup WHERE k BETWEEN 4850578 AND 8454295 OR k BETWEEN 8776291 AND 29749077;"
Benchmark
	Average number of seconds to run all queries: 0.041 seconds
	Minimum number of seconds to run all queries: 0.037 seconds
	Maximum number of seconds to run all queries: 0.115 seconds
	Number of clients running queries: 10
	Average number of queries per client: 1
```

after：
```
mysqlslap -hd3 -uroot -P9030  --create-schema=test_db -c 10 -i 500 -q "SELECT count(k) FROM sbtest1 WHERE k BETWEEN 4850578 AND 8454295 OR k BETWEEN 8776291 AND 29749077;"
Benchmark
	Average number of seconds to run all queries: 0.029 seconds
	Minimum number of seconds to run all queries: 0.027 seconds
	Maximum number of seconds to run all queries: 0.034 seconds
	Number of clients running queries: 10
	Average number of queries per client: 1
```
Problem Summary:

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [x] No need to test or manual test. Explain why:
        - [x] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [x] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by Thearas** [comment]
Thank you for your contribution to Apache Doris.
Don't know what should be done next? See [How to process your PR](https://cwiki.apache.org/confluence/display/DORIS/How+to+process+your+PR).

Please clearly describe your PR:
1. What problem was fixed (it's best to include specific error reporting information). How it was fixed.
2. Which behaviors were modified. What was the previous behavior, what is it now, why was it modified, and what possible impacts might there be.
3. What features were added. Why was this function added?
4. Which code was refactored and why was this part of the code refactored?
5. Which functions were optimized and what is the difference before and after the optimization?

**Comment by HappenLee** [comment]
run buildall

**Review by BiteTheDDDDt — DISMISSED** [review]

File: be/src/vec/exprs/vcompound_pred.h [hunk 1]
```diff
@@ -158,12 +158,15 @@ class VCompoundPred : public VectorizedFnCall {
         if (_can_fast_execute && fast_execute(context, block, result_column_id)) {
             return Status::OK();
         }
-        if (get_num_children() == 1 || !_all_child_is_compound_and_not_const()) {
+        if (get_num_children() == 1 || !_all_child_is_not_const()) {
```
Comment (zhiqiang-hhhh) [inline be/src/vec/exprs/vcompound_pred.h:161]: `!_all_child_is_not_const()` is hard to understand. It is equal to `has_const_child()`

File: be/src/vec/exprs/vcompound_pred.h [hunk 2]
```diff
@@ -342,10 +380,9 @@ class VCompoundPred : public VectorizedFnCall {
         return (l_null & r_null) | (r_null & (r_null ^ a)) | (l_null & (l_null ^ b));
     }
 
-    bool _all_child_is_compound_and_not_const() const {
-        return std::ranges::all_of(_children, [](const VExprSPtr& arg) -> bool {
-            return arg->is_compound_predicate() && !arg->is_constant();
-        });
+    bool _all_child_is_not_const() const {
```
Comment (zhiqiang-hhhh) [inline be/src/vec/exprs/vcompound_pred.h:383]: As we use this method like `!_all_child_is_not_const`, and it is equal to `_has_const_child`, so I think we should implement ``
`_has_const_child`, which is more easy to understand.

File: be/src/vec/exprs/vcompound_pred.h [hunk 3]
```diff
@@ -342,10 +380,9 @@ class VCompoundPred : public VectorizedFnCall {
...
+        return std::ranges::all_of(
```
Comment (zhiqiang-hhhh) [inline be/src/vec/exprs/vcompound_pred.h:384]:
```
return std::ranges::any_of(_children, [](const VExprSPtr& arg) -> bool {
            return arg->is_constant();
        });
```

File: be/src/vec/exprs/vcompound_pred.h (hunk 1 above)
Comment (zhiqiang-hhhh) [inline be/src/vec/exprs/vcompound_pred.h:161]: In what situation we will have just one child?

Comment (HappenLee) [reply be/src/vec/exprs/vcompound_pred.h:161]: `not` predicate

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Review by zhiqiang-hhhh — APPROVED** [review]
LGTM

**Review by zclllyybb — APPROVED** [review]
LGTM

**Comment by HappenLee** [comment]
run buildall

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**